compiler/src/iree/compiler/Codegen/Common/BlockDynamicDimensions.cpp

```diff
@@ -154,18 +158,23 @@ blockDynamicDimensionsOfValue(RewriterBase &rewriter,
   auto outputType = RankedTensorType::get(
       staticOutputShape, tensorType.getElementType(), tensorType.getEncoding());
 
-  Value expandShape = rewriter.create<tensor::ExpandShapeOp>(
+  auto expandShapeOp = rewriter.create<tensor::ExpandShapeOp>(
       loc, outputType, v, reassociation, outputShape);
-  Value barrier =
-      rewriter.create<IREE::Util::OptimizationBarrierOp>(loc, expandShape)
-          .getResult(0);
-  Value collapseShape = rewriter.create<tensor::CollapseShapeOp>(
+  Value barrier = rewriter
+                      .create<IREE::Util::OptimizationBarrierOp>(
+                          loc, expandShapeOp.getResult())
+                      .getResult(0);
+  auto collapseShapeOp = rewriter.create<tensor::CollapseShapeOp>(
       loc, tensorType, barrier, reassociation);
-  return collapseShape;
+  return ReshapeOps{expandShapeOp, collapseShapeOp};
```

Comment:
can we just do `return {expandShapeOp, collapseShapeOp}`?

Reply:
The type deduction fails when returning a `std::optional`. So I need to make it explicit.